cli/cli.ts

```diff
@@ -644,26 +638,93 @@ function tagReleaseAsync(parsed: commandParser.ParsedCommand) {
         })
 }
 
-function bumpAsync(parsed?: commandParser.ParsedCommand) {
+async function bumpAsync(parsed?: commandParser.ParsedCommand) {
     const bumpPxt = parsed && parsed.flags["update"];
     const upload = parsed && parsed.flags["upload"];
+    let pr = parsed && parsed.flags["pr"];
+    let bumpType = parsed && parsed.flags["version"] as string;
+    if (!bumpType) bumpType = "patch";
+
+    let currBranchName = "";
+    let token = "";
+    let user = "";
+    let owner = "";
+    let repo = "";
+    let branchProtected = false;
+
+    try {
+        currBranchName = await nodeutil.getCurrentBranchNameAsync();
+        token = await nodeutil.getGitHubTokenAsync();
+        user = await nodeutil.getGitHubUserAsync(token);
+        ({ owner, repo } = await nodeutil.getGitHubOwnerAndRepoAsync());
+        branchProtected = await nodeutil.isBranchProtectedAsync(token, owner, repo, currBranchName);
+        if (branchProtected) {
+            console.log(`${FgYellow}Branch ${currBranchName} is protected; creating a pull request instead of pushing directly.${Reset}`);
+        }
+    } catch (e) {
+        console.warn(`${FgYellow}Unable to determine branch protection status.${Reset}`, e.message);
+    }
+
     if (fs.existsSync(pxt.CONFIG_NAME)) {
         if (upload) throw U.userError("upload only supported on targets");
 
-        return Promise.resolve()
-            .then(() => nodeutil.runGitAsync("pull"))
-            .then(() => justBumpPkgAsync())
-            .then(() => nodeutil.runGitAsync("push", "--tags"))
-            .then(() => nodeutil.runGitAsync("push"))
+        if (pr || branchProtected) {
```

Comment:
async/await?

Reply:
I typically don't proactively convert to async/await and just do it when it simplifies the changes I'm making. The need didn't arise here. Happy to do it if you'd prefer that!

Reply:
I also wanted to leave it in the promise format to clearly show that the non-pr codepath works the same as before.